java/src/main/java/com/powsybl/python/network/NetworkModificationsCFunctions.java

```diff
@@ -199,4 +202,95 @@ public static void createBranchFeederBaysTwt(IsolateThread thread, ObjectHandle
             modification.apply(network);
         });
     }
+
+    @CEntryPoint(name = "getConnectablesOrderPositions")
+    public static PyPowsyblApiHeader.ArrayPointer<PyPowsyblApiHeader.SeriesPointer> getConnectablesOrderPositions(IsolateThread thread, ObjectHandle networkHandle,
+                                                                                                                  CCharPointer voltageLevelId, PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            String voltageLevelIdStr = CTypeUtil.toString(voltageLevelId);
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+
+            VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelIdStr);
+            Map<String, List<ConnectablePosition.Feeder>> feederPositionsOrders = getFeedersByConnectable(voltageLevel);
+            return Dataframes.createCDataframe(Dataframes.feederMapMapper(), feederPositionsOrders);
+        });
+
+    }
+
+    @CEntryPoint(name = "getUnusedConnectableOrderPositions")
+    public static PyPowsyblApiHeader.ArrayPointer<CIntPointer> getUnusedConnectableOrderPositions(IsolateThread thread, ObjectHandle networkHandle,
+                                                                                                  CCharPointer busbarSectionId, CCharPointer beforeOrAfter,
+                                                                                                  PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+            String busbarSectionIdStr = CTypeUtil.toString(busbarSectionId);
+            BusbarSection busbarSection = network.getBusbarSection(busbarSectionIdStr);
+            Optional<Range<Integer>> positionsOrders;
+            if (CTypeUtil.toString(beforeOrAfter).equals("BEFORE")) {
+                positionsOrders = getUnusedOrderPositionsBefore(busbarSection);
+            } else {
+                positionsOrders = getUnusedOrderPositionsAfter(busbarSection);
+            }
+            if (positionsOrders.isPresent()) {
+                int max = positionsOrders.get().getMaximum();
+                int min = positionsOrders.get().getMinimum();
+                return createIntegerArray(Arrays.asList(min, max));
+            } else {
+                return createIntegerArray(Collections.EMPTY_LIST);
+            }
+        });
+
+    }
+
+    static Map<String, List<ConnectablePosition.Feeder>> getFeedersByConnectable(VoltageLevel voltageLevel) {
+        Map<String, List<ConnectablePosition.Feeder>> feedersByConnectable = new HashMap<>();
```

Comment:
Very minor detail too: note that the hash map will introduce its own sort order:
it could be more natural to keep the order of the connectables stream?